crates/sui-indexer-alt-framework/src/lib.rs

```diff
@@ -235,22 +246,49 @@ impl Indexer {
         };
 
         if self.skip_watermark {
+            // If skip_watermark is enabled, there will be inconsistency between the watermark and the
+            // data in the database. This is generally not desired, but useful for benchmarks.
+            // To accommodate this, we allow it to be passed only if there is no initial watermark
+            // in the database, which is a good indicator that we are running a benchmark.
+            ensure!(
+                watermark.is_none(),
+                "You can only skip watermarks for sequential pipelines if there is no initial watermark in the database"
+            );
             warn!(
                 pipeline = H::NAME,
                 "--skip-watermarks enabled and ignored for sequential pipeline"
             );
+        } else {
+            // Note that if there is no initial watermark, first_checkpoint must be 0 or non-existent.
+            let expected_first_checkpoint = watermark
+                .as_ref()
+                .map(|w| w.checkpoint_hi_inclusive as u64 + 1)
+                .unwrap_or_default();
+            // For a sequential pipeline, data must be written in the order of checkpoints.
+            // Hence, we do not allow the first_checkpoint override to be in arbitrary positions.
+            ensure!(
+                expected_first_checkpoint == self.first_checkpoint.unwrap_or_default(),
```

Comment:
This is doing something different from the previous suggestion -- is that intentional? The suggestion was that if there is no initial watermark, it's fine to supply a first checkpoint, and use that to seed both the next checkpoint and the initial watermark, even if we are not skipping the watermark; and if there is an initial watermark, then the first checkpoint is still allowed to be less than or equal to it (and the sequential pipeline will just wait for its next checkpoint).

It seems like here, if there is no initial watermark, we're going to require that `--first-checkpoint` is `0`, and if there is an initial watermark, then `--first-checkpoint` has to match it exactly. This is going to make it quite difficult to use the `--first-checkpoint` on an instance of the indexer that combines a sequential pipeline with something else (imagine it's paired with a concurrent pipeline that has fallen behind or needs to be backfilled).

I think the code suggestion (along with passing `next_checkpoint` down into the `committer` task) would implement the suggestion -- was there an issue with it?